CMakeLists.txt

```diff
@@ -48,9 +48,7 @@ set_package_properties(Boost PROPERTIES
     URL "https://www.boost.org"
     PURPOSE "Used throughout OMPL for data serialization, graphs, etc.")
 set(Boost_USE_MULTITHREADED ON)
-set(Boost_NO_BOOST_CMAKE ON)
-find_package(Boost 1.58 QUIET REQUIRED COMPONENTS serialization filesystem system program_options)
-include_directories(SYSTEM ${Boost_INCLUDE_DIR})
```

Comment:
Why is this line deleted? If I have set the environment variable `BOOST_ROOT` to some non-standard boost installation, will the correct include path be used?

Reply:
The includes are brought in through the call to `target_link_libraries`. The exported namespace targets that are found above already have the properties set. 

According to the [docs](https://cmake.org/cmake/help/latest/module/FindBoost.html#hints), it looks like `BOOST_ROOT` is used by the module as a hint of where to find boost. Therefore, this would be my expected behavior if you have installed boost in a custom location.
1. Cmake tries to find it boostConfig.cmake from find_package and fails since it's not on cmake prefix path
2. You gave a hint for the module to find boost from `BOOST_ROOT`
3. The find succeeds, and populates the imported namespace targets
4. `target_link_libraries` functions the same.

If you have a system without boost installed in system headers and instead in a non-standard location, please try. 

If you think this is a common way people find boost, perhaps one could add it to CI?

Reply:
> The includes are brought in through the call to target_link_libraries. The exported namespace targets that are found above already have the properties set.

Neat! I learnt something today. I tried it and it works with boost in a non-standard location.